core/lib/multistore.ts

```diff
@@ -246,17 +254,14 @@ export function localizedRoute (routeObj: LocalizedRoute | string | RouteConfig
   return routeObj
 }
 
-export function setupMultistoreRoutes (config, router: VueRouter, routes: RouteConfig[]): void {
-  const allStoreRoutes = [...routes]
-  if (config.storeViews.mapStoreUrlsFor.length > 0 && config.storeViews.multistore === true) {
-    for (const storeCode of config.storeViews.mapStoreUrlsFor) {
-      if (storeCode && (config.defaultStoreCode !== storeCode)) {
-        for (const route of routes) {
-          const localRoute = localizedRoute(Object.assign({}, route), storeCode)
-          allStoreRoutes.push(localRoute)
-        }
-      }
-    }
+export function setupMultistoreRoutes (config, router: VueRouter, routes: RouteConfig[], priority: number = 0): void {
+  const allRoutes = []
+  const isMultistore = config.storeViews.multistore === true && config.storeViews.mapStoreUrlsFor.length > 0
+  const { storeCode } = currentStoreView()
+  if (isMultistore && storeCode) {
+    allRoutes.push(...routes.map(route => localizedRoute(Object.assign({}, route), storeCode)))
+  } else {
+    allRoutes.push(...routes)
   }
-  RouterManager.addRoutes(allStoreRoutes, router)
+  RouterManager.addRoutes(allRoutes, router, true, priority)
```

Comment:
Oh, I've seen hyou already optimized it :)

Reply:
Yeah, I removed it from every case :) 
It's easier and simpler without the name prefix and it works correctly in all the cases I tested